### PR TITLE
Demo Account Routing

### PR DESCRIPTION
**Pull request by vecchp**
OUT-95

Create an account locally in your django app with an `@example.com` email address.  Add that user to a org.  Log in with it and you should to local dev.  Log out and you should be able to log in with social in prod (or using a different username + pass that isn't an example.com address).  You will likely need to pull the latest simulator builds so that the async storage library is included.

https://github.com/user-attachments/assets/5f1a4b71-3dd3-4ead-9133-67a36d5609fb

### CODE REVIEW

File: apps/betterangels/config.ts [hunk 1]
```diff
@@ -5,26 +5,28 @@ import { Platform } from 'react-native';
...
   const apiUrl = process.env.EXPO_PUBLIC_API_URL;
+  const demoApiUrl = process.env.EXPO_PUBLIC_DEMO_API_URL;
   const redirectUri =
     Platform.OS === 'web'
       ? makeRedirectUri()
       : process.env.EXPO_PUBLIC_REDIRECT_URL;
 
   // Check if any of the environment variables are undefined
-  if (!googleClientId || !apiUrl || !redirectUri) {
+  if (!googleClientId || !apiUrl || !redirectUri || !demoApiUrl) {
```
Comment (vecchp) [inline apps/betterangels/config.ts:15]: Put this next to apiurl

File: apps/betterangels/eas.json [hunk 2]
```diff
@@ -43,7 +46,7 @@
       "ios": {
         "ascAppId": "6468135540"
       },
-     "android": {
+      "android": {
```
Comment (vecchp) [inline apps/betterangels/eas.json:49]: no space?

File: libs/expo/shared/apollo/src/lib/client.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    }
+  };
+
+  return (
+    <ApolloClientContext.Provider
+      value={{
+        client,
+        setProductionApi,
+        setDemoApi,
+        currentApiUrl, // Provide the current API URL in the context
```
Comment (vecchp) [inline libs/expo/shared/apollo/src/lib/client.tsx:102]: Probably not needed?

File: libs/expo/shared/apollo/tsconfig.json [hunk 4]
```diff
@@ -2,12 +2,14 @@
...
     "module": "commonjs",
+    "esModuleInterop": true,
     "forceConsistentCasingInFileNames": true,
     "strict": true,
     "noImplicitOverride": true,
     "noPropertyAccessFromIndexSignature": true,
     "noImplicitReturns": true,
-    "noFallthroughCasesInSwitch": true
+    "noFallthroughCasesInSwitch": true,
+    "jsx": "react"
```
Comment (vecchp) [inline libs/expo/shared/apollo/tsconfig.json:12]: We now have `tsx` in the lib so it appears that `esModuleInterop` and `jsx` is now necessary.

**Comment by vecchp** [comment]
In progress note:

This is a rough sketch of what I'm trying to achieve.  What I am running into now is that once the dev api url is set it seems to be falling back to the production api??

```
› Press ? │ show all commands
iOS Bundled 170ms index.ts (1 module)
 // App Open
 LOG  URL: https://api.dev.betterangels.la
 LOG  URL: https://api.dev.betterangels.la
 LOG  URL: https://api.dev.betterangels.la
 // Logging in with demo account
 LOG  Username: demoaccount@example.com
 LOG  Set demo api
 // Confusing loop
 LOG  URL: https://api.dev.betterangels.la   // why does this get called
 LOG  URL: http://localhost:8000 // this is what we actually want
 LOG  URL: https://api.dev.betterangels.la // why is this falling back to api dev?
```

**Review by mikefeldberg — APPROVED** [review]

**Pull request merged**